great_expectations/render/renderer/inline_renderer.py

```diff
@@ -121,68 +123,87 @@ def _get_atomic_rendered_content_from_renderer_names(
+            logger.info(error_message)
+
+            failure_renderer: AtomicPrescriptiveRendererType | AtomicDiagnosticRendererType
+            if renderer_name.startswith(AtomicRendererType.PRESCRIPTIVE):
+                failure_renderer = AtomicPrescriptiveRendererType.FAILED
+                failure_renderer_message = f'Renderer "{failure_renderer}" will be used to render prescriptive content.'
+            else:
+                failure_renderer = AtomicDiagnosticRendererType.FAILED
+                failure_renderer_message = f'Renderer "{failure_renderer}" will be used to render diagnostic content.'
+            logger.info(failure_renderer_message)
+
+            renderer_impl = get_renderer_impl(
+                object_name=expectation_type, renderer_type=failure_renderer
+            )
+            if renderer_impl:
                 renderer_rendered_content = (
-                    InlineRenderer._get_renderer_atomic_rendered_content(
+                    InlineRenderer._get_rendered_content_from_renderer_impl(
+                        renderer_impl=renderer_impl,
                         render_object=render_object,
-                        renderer_name=failed_renderer_type,
-                        expectation_type=expectation_type,
                     )
                 )
-                rendered_content.append(renderer_rendered_content)
+                renderer_rendered_content.exception = error_message
+            else:
+                raise InlineRendererError(
+                    f'Renderer "{failure_renderer}" was not found in the registry.'
+                )
 
-        return rendered_content
+        return renderer_rendered_content
 
     @staticmethod
-    def _get_renderer_atomic_rendered_content(
-        render_object: Union[ExpectationConfiguration, ExpectationValidationResult],
-        renderer_name: Union[
-            str, AtomicDiagnosticRendererType, AtomicPrescriptiveRendererType
-        ],
-        expectation_type: str,
+    def _get_rendered_content_from_renderer_impl(
+        renderer_impl: RendererImpl,
+        render_object: ExpectationConfiguration | ExpectationValidationResult,
     ) -> RenderedAtomicContent:
-        renderer_impl: Optional[RendererImpl] = get_renderer_impl(
-            object_name=expectation_type, renderer_type=renderer_name
-        )
-        if renderer_impl:
-            renderer_fn: Callable[
-                ..., Union[RenderedAtomicContent, RenderedContent]
-            ] = renderer_impl.renderer
-            if isinstance(render_object, ExpectationConfiguration):
-                renderer_rendered_content = renderer_fn(configuration=render_object)
-            else:
-                renderer_rendered_content = renderer_fn(result=render_object)
+        renderer_fn: Callable[
+            ..., RenderedAtomicContent | RenderedContent
```

Comment:
If this gets reused a lot you could save this variable as a `TypeAlias`

```python
RendererFn: TypeAlias =  Callable[..., RenderedAtomicContent | RenderedContent
```

Reply:
It doesn't get reused often and I'd prefer to not declare this a type since the goal is to deprecate legacy `RenderedContent`.